librz/util/zip.c

```diff
@@ -266,3 +266,79 @@ RZ_API bool rz_deflatew_buf(RZ_NONNULL RzBuffer *src, RZ_NONNULL RzBuffer *dst,
 
 	return true;
 }
+
+/**
+ * \brief inflate compressed data in RzBbuffer, use MAX_WBITS as the window size logarithm.
+ * \see rz_inflatew_buf()
+ */
+RZ_API bool rz_inflate_buf(RZ_NONNULL RzBuffer *src, RZ_NONNULL RzBuffer *dst, ut64 block_size, ut8 *src_consumed, ut8 *dst_len) {
+	rz_return_val_if_fail(src && dst, false);
+	return rz_inflatew_buf(src, dst, block_size, src_consumed, dst_len, MAX_WBITS + 16);
+}
+
+/**
+ * \brief inflate data contained in a RzBuffer using zlib
+ * \param src source buffer
+ * \param dst destination buffer
+ * \param block_size block sizes to use while inflating data
+ * \param src_consumed consumed source buffer length
+ * \param dst_len inflated buffer length
+ * \param wbits the size of the history buffer (or “window size”), and what header and trailer format is expected.
+ * \return true if successful; false otherwise
+ */
+RZ_API bool rz_inflatew_buf(RZ_NONNULL RzBuffer *src, RZ_NONNULL RzBuffer *dst, ut64 block_size, ut8 *src_consumed, ut8 *dst_len, int wbits) {
+	rz_return_val_if_fail(src && dst, false);
+
+	int err = 0, flush = Z_NO_FLUSH;
+	ut64 dst_cursor = 0, src_cursor = 0;
+	ut64 src_readlen = 0;
+	z_stream stream;
+
+	if (block_size <= 0) {
+		return false;
+	}
+
+	memset(&stream, 0, sizeof(z_stream));
+
+	stream.zalloc = Z_NULL;
+	stream.zfree = Z_NULL;
+	stream.opaque = Z_NULL;
+
+	if (inflateInit2(&stream, wbits) != Z_OK) {
+		return false;
+	}
+
+	int comp_factor = 1032; // maximum compression ratio
```

Comment:
As of now, I just prepare for the highest possible compression ratio and assign memory accordingly, but this forces us not choose a large enough `block_size`. So, we either need to go with a small `block_size` (right now, I am using `1 << 13 = 8 KB` (https://github.com/rizinorg/rizin/pull/1446/commits/b3fa0d5630f65a0deab3d1d08c75e34e845485aa#diff-b6e47a6b965ef83d50289bc0568bb5f82a4e47d5fe654cc6be44a2c85c7b0ce8R1309), which is kind of small for large files, but still amounts to 9 MB memory) and subsequently more number of inflate loops, or go with a larger block size, thus larger memory usage. If there's any better way to handle this, please feel free to suggest.